azure-core/azure/core/pipeline/transport/aiohttp.py

```diff
@@ -49,21 +50,28 @@ class AioHttpTransport(AsyncHttpTransport):
     """AioHttp HTTP sender implementation.
     """
 
-    def __init__(self, configuration=None, *, loop=None):
+    def __init__(self, configuration=None, *, session=None, loop=None, external_owner=False):
         self._loop = loop
-        self.session = None
-        self.config = configuration
+        self._external_owner = external_owner
+        self.session = session
+        self.config = configuration or Configuration()
 
```

Comment:
Not
self._external_owner = false
if session is not none:
    self._external_owner = true
?

Reply:
Just because a user passes in a custom session doesn't mean they want to own it.
However I'm not sure what the correct default here would be....

Reply:
My hypothesis is that there are two primary reasons as to why someone would pass in a custom session:

- I want to configure my session in some way that the pipeline/configuration/policies don't allow for
- I want to share the session between multiple clients

My assumption is that the primary reason for someone to pass in a session is the latter (I'm assuming we provide enough configuration knobs for the vast majority of scenarios through configuration properties). If we all agree on this, let's optimize the API for that scenario.

Reply:
@johanste - While I agree that makes sense in terms of the appropriate default when a user passes in a session - it is not the default behaviour when they don't pass in a session.

That means that the default value documented in the constructor is not actually representative of the default transport behaviour *unless* someone passes in a session.

That seems a touch odd to me - however it's probably fine given that an end user is unlikely to construct a Transport unless they want to pass in a session....

Reply:
It also means that we have to add special logic to know if the session we are using has been passed into us or whether it was created by the Transport.
Be defaulting ownership to the Transport, we don't have to care where the session came from.